pkg/kubelet/kubelet_test.go

```diff
@@ -2586,6 +2586,7 @@ func TestHandlePodResourcesResize(t *testing.T) {
 	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, true)
 	testKubelet := newTestKubelet(t, false)
 	defer testKubelet.Cleanup()
+	metrics.Register()
```

Comment:
Your unit test execution fails because metrics.Register() only executes once per test run.
ref: https://github.com/kubernetes/kubernetes/blob/f59dd4bce33f454e5c9386be56fa3d11e3bb2d7a/pkg/kubelet/metrics/metrics.go#L1013-L1019

When you test a single test case, it works fine because in a single test, the feature gate InPlacePodVerticalScaling is enabled first, and then metrics.Register() is executed.

However, in an overall test run, metrics.Register() was already executed earlier, so it will not be executed again in your test function.

Although we can't use metrics.Register(), we can still register individual metrics. Here’s a reference code snippet:
```
registry := k8smetrics.NewKubeRegistry()
if err := registry.Register(metrics.ResizeRequestsTotal); err != nil {
    t.Errorf("unexpected error: %v", err)
}

// ...

if tt.expectMetricCount != 0 {
    expectedMetrics := fmt.Sprintf(`
    # HELP kubelet_resize_requests_total [ALPHA] Total number of resize requests observed by the Kubelet.
    # TYPE kubelet_resize_requests_total counter
    kubelet_resize_requests_total{resize_state="%v"} %v
    `, tt.expectedResize, tt.expectMetricCount)
    if err := testutil.GatherAndCompare(registry, strings.NewReader(expectedMetrics), "kubelet_resize_requests_total"); err != nil {
        t.Error(err)
    }
}
```
This should solve your issue. Let’s resolve this and rebase it.